client/js/lounge.js

```diff
@@ -994,6 +994,23 @@ $(function() {
 		});
 	});
 
+	$("#connect").one("show", function() {
+		var params = window.URI(document.location.search);
+		params = params.search(true);
+		// Possible parameters:  name, host, port, password, tls, nick, username, realname, join
+		for (var key in params) {
+			var value = params[key];
+			key = key.replace(/\W/g, "");
+
+			var element = $("#connect input[name='" + key + "']");
```

Comment:
This also needs to check that:

- input is not `disabled` (when `lockNetwork` is `true`)
- input is not hidden (when `displayNetwork` is `false`). This might require a bit more fiddling than the one above, but it would be pretty nasty to let someone send you to a connect window with a bad host that you cannot see anywhere.